doc/tutorials/introduction/test_writing/accuracy_test_writing.markdown

```diff
@@ -0,0 +1,269 @@
+- (*Linux*)
+
+If you want to run **CERTAIN TESTS** (SUM.int test, for example).
+
+- If you work in IDE you need to find the project preference named `Command Arguments` or similar and write there the key like `--gtest_filter=<KEY>` (KEY = SUM.int in our example).  [gtest_filter.doc](--gtest_filter)
+- If you want to use your system:
+  - (*Windows*) move to the folder `<BUILD_DIR>/bin/Debug(Release)/` and run the executive file in the console `opencv_test_<MODULE_NAME>.exe --gtest_filter=<KEY>` (KEY = SUM.int in our example).
+  - (*Linux*)
+
+
+
+# TEST DATA
+
+Repository with the test data (opencv_extra): https://github.com/opencv/opencv_extra
+
+Before working with data you need to set environment variable `OPENCV_TEST_DATA_PATH`:
+
+- (Windows)
+  1. Win+R
+  2. Sysdm.cpl
+  3. Go to Advanced -> Environment Variables -> System variables (New..)
+  4. Set name as `OPENCV_TEST_DATA_PATH` and value as `<PATH_TO_opencv_extra>/testdata`
+- (Visual Studio)
+  1.  Go to Project -> Properties -> Debugging -> Environment
+  2. Set the Environment as `OPENCV_TEST_DATA_PATH=<PATH_TO_opencv_extra>/testdata`
+- (Linux)
+
+
+
+# ASSERTION
+
+**Assertion documentation**:
+
+- Link: https://github.com/google/googletest/blob/master/googletest/docs/primer.md
+
+
+
+# --gtest_filter
+
+Usage: `--gtest_filter=<KEY>`
+
+This KEY is the logical expression, which help to find needful tests names.
+
+**\*** - logical operator, which includes all symbols before or after this expression.
+
+​	**Examples:** (*smth* - any variation of symbols.)
+
+​	`--gtest_filter=smth*<key_word>`
+
+​	`--gtest_filter=smth*<key_word>*smth`
+
+​	`--gtest_filter=<key_word>*smth`
+
+**:** - logical operator, which concatenate logical expressions.
+
+​	**Examples:** (log_exp - logical expresson.)
+
+​	`--gtest_filter=log_exp1:log_exp`
+
+**Example:**
+
+Tests names:
+
+- TEST_sum.int
+- TEST_sum.float
+- TEST_sub.int
+- TEST_sub.float
+
+Logical keys:
+
+1. \*sum.\*
+2. TEST_sum.i\*
+3. \*sum.float
+4. TEST_sum.i\*:TEST_sub.\*
+
+Results:
+
+1. TEST_sum.int, TEST_sum.float
+2. TEST_sum.int
+3. TEST_sum.float
+4. TEST_sum.int, TEST_sub.int, TEST_sub.float
+
+
+
+# --gtest_param_filter
+
+Usage: `--gtest_params_filter=(<PARAM1>, <PARAM1>, ...)`
+
+This sentence helps to pass extra parameters to functions.
+
+For a getting the parameters you need to use `get<i>(GetParam())` or `GET_PARAM(i)`, where **i**
+
+is the index of the parameter.
+
+
+
+# BuildBot
+
+OpenCV's continuous integration system is available here:
+
+- [http://pullrequest.opencv.org](http://pullrequest.opencv.org/)
+- [http://pullrequest.opencv.org/contrib](https://pullrequest.opencv.org/#/summary/contrib)
+
+This system check code style, build all OpenCV and run tests. You can read more about this bot [here](https://pullrequest.opencv.org/buildbot/).
+
+The BuildBot page looks like a grid where rows are the `pull requests` and columns are `id, author, Title & description, Docs, Builds `.
```

Comment:
incorrect: this page is not part of BuildBot.

This is a dedicated dashboard for GitHub Pull Request (or GitLab Merge Requests).

Reply:
done